src/serverenvironment.cpp

```diff
@@ -1579,6 +1596,14 @@ void ServerEnvironment::step(float dtime)
 	// Send outdated detached inventories
 	m_server->sendDetachedInventories(PEER_ID_INEXISTENT, true);
 
+	// Notify mods of modified mapblocks
+	if (m_on_mapblocks_changed_receiver.receiving &&
+			!m_on_mapblocks_changed_receiver.modified_blocks.empty()) {
+		std::unordered_set<v3s16> modified_blocks;
```

Comment:
why we do a swap here and not just pass a const ref ?

Reply:
The internal set of modified blocks has to be cleared before calling the Lua callbacks so it is ready to record any changes made by them. The set could be cleared by the `on_mapblocks_changed` function, but this API is cleaner.